program/freedom-metal-program.wake

```diff
@@ -1,4 +1,28 @@
 # FIXME
+def addMetalMemory type memSubnode inDTSContents =
```

Comment:
Since this function is doing something reasonably complex, could you add a comment describing at a high level what it's doing?

My best guess is that given a device tree property type name and a node type, it finds the label of the matching node, and then it injects a reference to that label to the `chosen` node.